### PR TITLE
Use AdvancedMarkerElements and add index to markers on map

### PR DESCRIPTION
**Pull request by amysorto**
Part of #1477. Adds annotations for submission markers (but not polygons). @gino-m As far as I know we don't have polygons yet in our submissions. Should this PR close this issue instead and we file a separate issues for polygons?

I also filed #2108 to use custom HTML and CSS for the markers. When I initially tried the styles were not being correctly applied so I figured I can take a look into that separately so we can have better map annotiations in the meanwhile.
 
Screenshot:
Screenshot 2024-11-27 at 10 37 28 AM 1

Also I changed the index in the submission panel to start at 1 instead of 0. I thought there was a filed issue for this but I couldn't find it to link it here.

### CODE REVIEW

File: web/src/app/pages/main-page-container/main-page/map/map.component.ts [hunk 1]
```diff
@@ -137,52 +138,56 @@ export class MapComponent implements AfterViewInit, OnChanges, OnDestroy {
...
         this.lois$,
         this.navigationService.getLocationOfInterestId$(),
+        this.navigationService.getTaskId$(),
         this.selectedJob$,
-      ]).subscribe(([survey, lois, locationOfInterestId, selectedJob]) => {
-        const loisMap = ImmutableMap(
-          lois
-            .filter(loi =>
-              this.showPredefinedLoisOnly ? loi.predefined : true
+      ]).subscribe(
```
Comment (gino-m) [inline web/src/app/pages/main-page-container/main-page/map/map.component.ts:143]: This method is becoming unwieldy. While you're here any chance you could refactor it a bit, perhaps pulling out meaningful logic into helper functions?

**Review by gino-m — CHANGES_REQUESTED** [review]
Thanks, @amysorto! Lgtm, one small code health tax if you have time. 

@rfontanarosa for  final approval.

**Comment by gino-m** [comment]
@amysorto Please ping when RFAL!

**Comment by amysorto** [comment]
@gino-m This PR is ready for review!

**Pull request merged**